saltedge-compliance-connector-java-v2/compliance-connector-sdk/src/main/java/com/saltedge/connector/sdk/provider/ConnectorSDKCallbackService.java

```diff
@@ -187,13 +188,22 @@ public String onPaymentInitiationAuthorizationSuccess(
     Map<String, String> paymentExtraMap = parseExtra(paymentExtra);
 
     String sessionSecret = paymentExtraMap.get(SDKConstants.KEY_SESSION_SECRET);
-    String status = getStatusOfPaymentProduct(paymentProduct);
+    String status = getFinalStatusOfPaymentProduct(paymentProduct);
     SessionSuccessCallbackRequest params = new SessionSuccessCallbackRequest(userId, status);
     if (!StringUtils.isEmpty(sessionSecret)) sessionsCallbackService.sendSuccessCallback(sessionSecret, params);
 
     return paymentExtraMap.getOrDefault(SDKConstants.KEY_RETURN_TO_URL, "");
   }
 
+  @Override
+  public void updatePaymentFundsInformation(Boolean fundsAvailable, String paymentExtra, String status) {
+    Map<String, String> paymentExtraMap = parseExtra(paymentExtra);
+    String sessionSecret = paymentExtraMap.get(SDKConstants.KEY_SESSION_SECRET);
+
+    SessionUpdateCallbackRequest updateParams = new SessionUpdateCallbackRequest(fundsAvailable, status);
+    if (sessionSecret != null) sessionsCallbackService.sendUpdateCallback(sessionSecret, updateParams);
```

Comment:
Please add the same check of `sessionSecret` for all `sessionsCallbackService` methods, where applicable.